pkg/experiment/metastore/metastore_state_add_block.go

```diff
@@ -23,6 +25,9 @@ func (m *Metastore) AddBlock(_ context.Context, req *metastorev1.AddBlockRequest
 	_ = level.Info(l).Log("msg", "adding block")
 	t1 := time.Now()
 	defer func() {
+		if err == nil {
+			m.placementMgr.RecordStats(statSamplesFromMeta(req.Block))
```

Comment:
Nit, perhaps we want to abstract this call here, we probably don't want "add blocks" to be able to start/stop the placement manager.

Reply:
Could you please elaborate? I'd like to avoid abstractions introduced for sake of abstraction.

> we probably don't want "add blocks" to be able to start/stop the placement manager.

I don't think this is the case: `RecordStats` does not control the placement manager's state.

Reply:
However, I think I understand what you mean here.

The thing is that our metastore is quite cumbersome. I was going to split it into multiple services:

```proto

service MetastoreService {
  rpc AddBlock(AddBlockRequest) returns (AddBlockResponse) {}
}

service MetadataQueryService {
  rpc QueryMetadata(QueryMetadataRequest) returns (QueryMetadataResponse) {}
}

service TenantService {
  // Same as GetProfileStats
  rpc GetTenant(GetTenantRequest) returns (GetTenantResponse) {}
  rpc DeleteTenant(DeleteTenantRequest) returns (DeleteTenantResponse) {}
}

// Methods needed for the operator interface.
service RaftNodeService {
  rpc GetCommitIndex(GetCommitIndexRequest) returns (GetCommitIndexResponse) {}
  rpc ListPeers(ListPeersRequest) returns (ListPeersResponse) {}
}
```

In this way, `AddBlock` can be easily abstracted from the underlying metastore implementation:

```go
type StatsRecorder interface {
	RecordStats(*metastorev1.BlockMeta)
}

type RaftLog[T, R any] interface {
	Apply(context.Context, T) (R, error)
}

type MetastoreService struct {
	StatsRecorder
	RaftLog[*metastorev1.AddBlockRequest, *metastorev1.AddBlockResponse]
}
```

Same for other methods:
```go
type MetadataIndex interface {
	FindBlocksInRange(start, end int64, tenants []string) ([]*metastorev1.BlockMeta, error)
}

// To ensure consistent reads.
type RaftLeader interface {
	CommitIndex(context.Context) (uint64, error)
}

type MetadataQueryService struct {
	RaftLeader
	MetadataIndex
}
```

Reply:
Yeah, the underlying "issue" is that the add block command implementation is part of the `Metastore`, so it has access to many things it shouldn't have access to (including the entire `PlacementManager` interface). I don't like too many abstractions either, but a few (like the ones you suggested) can make things cleaner.

Reply:
We'll decompose the metastore; that will require quite a massive refactoring, though